### ISSUE TITLE
Keras Scattering1D example does not work with TF>=2.5

### ISSUE THREAD

**Issue by entn-at** [kymatio/kymatio#775]
When using Scattering1D with the Keras frontend (for example, in the spoken digit classification example,  `classif_keras.py`) of
 Kymatio 0.2.1 (installed from PyPi via pip) with TF 2.5 or 2.6, I'm getting the following error:
```ValueError: Error processing property 'phi_f' of DictWrapper(...```
TF2.4 works flawlessly. I'm not sure what change is causing this issue.

Minimal example for reproduction:
```
python3 -m venv ./venv
source venv/bin/activate
pip install tensorflow==2.6.0 kymatio=0.2.1
wget https://github.com/kymatio/kymatio/raw/master/examples/1d/classif_keras.py
python classif_keras.py
```

The full error log of running the `classif_keras.py` example script is below:
```
$ python classif_keras.py
2021-08-31 13:55:02.788089: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2021-08-31 13:55:02.788144: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
2021-08-31 13:55:15.781264: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory
2021-08-31 13:55:15.781343: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)
2021-08-31 13:55:15.781364: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (eduxps): /proc/driver/nvidia/version does not exist
2021-08-31 13:55:15.781694: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Model: "model"
_________________________________________________________________
Layer (type)                 Output Shape              Param #
=================================================================
input_1 (InputLayer)         [(None, 8192)]            0
_________________________________________________________________
2021-08-31 13:55:15.801286: W tensorflow/python/util/util.cc:348] Sets are not currently considered sequences, but this may change in the future, so consider avoiding using them.
Traceback (most recent call last):
  File "/testkymatio/venv/lib/python3.8/site-packages/tensorflow/python/module/module.py", line 369, in _flatten_module
    leaves = nest.flatten_with_tuple_paths(
  File "/testkymatio/venv/lib/python3.8/site-packages/tensorflow/python/util/nest.py", line 1644, in flatten_with_tuple_paths
    flatten(structure, expand_composites=expand_composites)))
  File "/testkymatio/venv/lib/python3.8/site-packages/tensorflow/python/util/nest.py", line 417, in flatten
    return _pywrap_utils.Flatten(structure, expand_composites)
TypeError: '<' not supported between instances of 'str' and 'int'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "classif_keras.py", line 179, in <module>
    model.summary()
  File "/testkymatio/venv/lib/python3.8/site-packages/keras/engine/training.py", line 2526, in summary
    layer_utils.print_summary(self,
  File "/testkymatio/venv/lib/python3.8/site-packages/keras/utils/layer_utils.py", line 254, in print_summary
    print_layer_summary(layers[i])
  File "/testkymatio/venv/lib/python3.8/site-packages/keras/utils/layer_utils.py", line 211, in print_layer_summary
    params = layer.count_params()
  File "/testkymatio/venv/lib/python3.8/site-packages/keras/engine/base_layer.py", line 2151, in count_params
    return layer_utils.count_params(self.weights)
  File "/testkymatio/venv/lib/python3.8/site-packages/keras/engine/base_layer.py", line 1343, in weights
    return self.trainable_weights + self.non_trainable_weights
  File "/testkymatio/venv/lib/python3.8/site-packages/keras/engine/base_layer.py", line 1310, in trainable_weights
    children_weights = self._gather_children_attribute('trainable_variables')
  File "/testkymatio/venv/lib/python3.8/site-packages/keras/engine/base_layer.py", line 2850, in _gather_children_attribute
    return list(
  File "/testkymatio/venv/lib/python3.8/site-packages/keras/engine/base_layer.py", line 2852, in <genexpr>
    getattr(layer, attribute) for layer in nested_layers))
  File "/testkymatio/venv/lib/python3.8/site-packages/tensorflow/python/module/module.py", line 175, in trainable_variables
    return tuple(
  File "/testkymatio/venv/lib/python3.8/site-packages/tensorflow/python/module/module.py", line 372, in _flatten_module
    six.raise_from(
  File "<string>", line 3, in raise_from
ValueError: Error processing property 'phi_f' of DictWrapper({0: array([1.        , 0.98786717, 0.9523448 , ..., 0.89595662, 0.9523448 ,
       0.98786717]), 1: array([0.5       , 0.49393359, 0.4761724 , ..., 0.44797831, 0.4761724 ,
       0.49393359]), 2: array([0.25      , 0.24696679, 0.2380862 , ..., 0.22398916, 0.2380862 ,
       0.24696679]), 3: array([0.125     , 0.1234834 , 0.1190431 , ..., 0.11199458, 0.1190431 ,
       0.1234834 ]), 4: array([0.0625    , 0.0617417 , 0.05952155, ..., 0.05599729, 0.05952155,
       0.0617417 ]), 5: array([3.12500000e-002, 3.08708491e-002, 2.97607750e-002, 2.79986444e-002,
       2.57055488e-002, 2.30310567e-002, 2.01371477e-002, 1.71822151e-002,
       1.43072926e-002, 1.16260683e-002, 9.21945801e-003, 7.13468852e-003,
       5.38817575e-003, 3.97105219e-003, 2.85605485e-003, 2.00458558e-003,
       1.37302918e-003, 9.17766154e-004, 5.98661620e-004, 3.81090307e-004,
       2.36739920e-004, 1.43519925e-004, 8.49082728e-005, 4.90213111e-005,
       2.76195721e-005, 1.51860937e-005, 8.14839952e-006, 4.26673490e-006,
       2.18029929e-006, 1.08726066e-006, 5.29112889e-007, 2.51281268e-007,
       1.16457912e-007, 5.26714152e-008, 2.32475979e-008, 1.00133245e-008,
       4.20896778e-009, 1.72651367e-009, 6.91132867e-010, 2.69991555e-010,
       1.02928566e-010, 3.82929452e-011, 1.39026864e-011, 4.92578884e-012,
       1.70313848e-012, 5.74673590e-013, 1.89229841e-013, 6.08072157e-014,
       1.90685521e-014, 5.83549130e-015, 1.74274659e-015, 5.07911637e-016,
       1.44457200e-016, 4.00947334e-017, 1.08600688e-017, 2.87061485e-018,
       7.40481796e-019, 1.86402165e-019, 4.57914744e-020, 1.09778024e-020,
       2.56828549e-021, 5.86365343e-022, 1.30644279e-022, 2.84059690e-023,
       6.02734327e-024, 1.24807117e-024, 2.52202789e-025, 4.97344736e-026,
       9.57110925e-027, 1.79748026e-027, 3.29429958e-028, 5.89195147e-029,
       1.02837696e-029, 1.75163100e-030, 2.91158864e-031, 4.72296269e-032,
       7.47646185e-033, 1.15498099e-033, 1.74120802e-034, 2.56167239e-035,
       3.67784670e-036, 5.15300741e-037, 7.04571412e-038, 9.40126677e-039,
       1.22417877e-039, 1.55560889e-040, 1.92909269e-041, 2.33454841e-042,
       2.75708261e-043, 3.17756016e-044, 3.57383825e-045, 3.92259174e-046,
       4.20153934e-047, 4.39178293e-048, 4.47992152e-049, 4.45961177e-050,
       4.33232279e-051, 4.10716045e-052, 3.79979038e-053, 3.43063659e-054,
       3.02264330e-055, 2.59893962e-056, 2.18073347e-057, 1.78569012e-058,
       1.42694327e-059, 1.11276749e-060, 8.46835892e-062, 6.28913898e-063,
       4.55806207e-064, 3.22378734e-065, 2.22510056e-066, 1.49875280e-067,
       9.85161611e-069, 6.31949042e-070, 3.95597690e-071, 2.41669888e-072,
       1.44074939e-073, 8.38207327e-075, 4.75895468e-076, 2.63674902e-077,
       1.42568350e-078, 7.52271421e-080, 3.87367436e-081, 1.94656434e-082,
       9.54578226e-084, 4.56826609e-085, 2.13347898e-086, 9.72349624e-088,
       4.32467665e-089, 1.87707635e-090, 7.95073620e-092, 3.28647139e-093,
       1.32571288e-094, 5.21874652e-096, 2.00484147e-097, 7.51607284e-099,
       2.74978684e-100, 9.81757406e-102, 3.42063276e-103, 1.16306991e-104,
       3.85924456e-106, 1.24967156e-107, 3.94899497e-109, 1.21779555e-110,
       3.66487604e-112, 1.07631974e-113, 3.08475324e-115, 8.62773226e-117,
       2.35488659e-118, 6.27249671e-120, 1.63045191e-121, 4.13592500e-123,
       1.02384549e-124, 2.47339397e-126, 5.83108375e-128, 1.34153612e-129,
       3.01198334e-131, 6.59933037e-133, 1.41105610e-134, 2.94432537e-136,
       5.99548647e-138, 1.19140704e-139, 2.31043099e-141, 4.37243085e-143,
       8.07513899e-145, 1.45537252e-146, 2.55973755e-148, 4.39353224e-150,
       7.35917793e-152, 1.20293427e-153, 1.91889670e-155, 2.98715944e-157,
       4.53797730e-159, 6.72764925e-161, 9.73333051e-163, 1.37422140e-164,
       1.89342912e-166, 2.54588337e-168, 3.34060451e-170, 4.27768430e-172,
       5.34551439e-174, 6.51879591e-176, 7.75786777e-178, 9.00978586e-180,
       1.02113620e-181, 1.12940572e-183, 1.21902722e-185, 1.28402632e-187,
       1.31987122e-189, 1.32399486e-191, 1.29609892e-193, 1.23818946e-195,
       1.15433846e-197, 1.05021045e-199, 9.32430803e-202, 8.07893282e-204,
       6.83106611e-206, 5.63663727e-208, 4.53888098e-210, 3.56676592e-212,
       2.73525317e-214, 2.04699866e-216, 1.49497763e-218, 1.06548892e-220,
       7.41071803e-223, 5.03000883e-225, 3.33176463e-227, 2.15365923e-229,
       1.35855357e-231, 8.36322256e-234, 5.02420845e-236, 2.94549813e-238,
       1.68518258e-240, 9.40875758e-243, 5.12642613e-245, 2.72580156e-247,
       1.41439553e-249, 7.16216995e-252, 3.53928452e-254, 1.70680316e-256,
       8.03245880e-259, 3.68901722e-261, 1.65336967e-263, 7.23146615e-266,
       3.08659648e-268, 1.28567282e-270, 5.22610553e-273, 2.07311310e-275,
       8.02536740e-278, 3.03182380e-280, 1.11773822e-282, 4.02136368e-285,
       1.41189925e-287, 4.83761357e-290, 1.61754267e-292, 5.27809770e-295,
       1.68072327e-297, 5.22290476e-300, 1.58389033e-302, 4.68743455e-305,
       1.35376236e-307, 3.81545870e-310, 1.04941723e-312, 2.81673978e-315,
       7.37806509e-318, 1.88584857e-320, 4.94065646e-323, 0.00000000e+000,
       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,
       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,
       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,
       0.00000000e+000, 0.00000000e+000, 0.00000000e+000, 0.00000000e+000,
       0.00000000e+000, 0.00000000e+000, 4.94065646e-323, 1.88584857e-320,
       7.37806509e-318, 2.81673978e-315, 1.04941723e-312, 3.81545870e-310,
       1.35376236e-307, 4.68743455e-305, 1.58389033e-302, 5.22290476e-300,
       1.68072327e-297, 5.27809770e-295, 1.61754267e-292, 4.83761357e-290,
       1.41189925e-287, 4.02136368e-285, 1.11773822e-282, 3.03182380e-280,
       8.02536740e-278, 2.07311310e-275, 5.22610553e-273, 1.28567282e-270,
       3.08659648e-268, 7.23146615e-266, 1.65336967e-263, 3.68901722e-261,
       8.03245880e-259, 1.70680316e-256, 3.53928452e-254, 7.16216995e-252,
       1.41439553e-249, 2.72580156e-247, 5.12642613e-245, 9.40875758e-243,
       1.68518258e-240, 2.94549813e-238, 5.02420845e-236, 8.36322256e-234,
       1.35855357e-231, 2.15365923e-229, 3.33176463e-227, 5.03000883e-225,
       7.41071803e-223, 1.06548892e-220, 1.49497763e-218, 2.04699866e-216,
       2.73525317e-214, 3.56676592e-212, 4.53888098e-210, 5.63663727e-208,
       6.83106611e-206, 8.07893282e-204, 9.32430803e-202, 1.05021045e-199,
       1.15433846e-197, 1.23818946e-195, 1.29609892e-193, 1.32399486e-191,
       1.31987122e-189, 1.28402632e-187, 1.21902722e-185, 1.12940572e-183,
       1.02113620e-181, 9.00978586e-180, 7.75786777e-178, 6.51879591e-176,
       5.34551439e-174, 4.27768430e-172, 3.34060451e-170, 2.54588337e-168,
       1.89342912e-166, 1.37422140e-164, 9.73333051e-163, 6.72764925e-161,
       4.53797730e-159, 2.98715944e-157, 1.91889670e-155, 1.20293427e-153,
       7.35917793e-152, 4.39353224e-150, 2.55973755e-148, 1.45537252e-146,
       8.07513899e-145, 4.37243085e-143, 2.31043099e-141, 1.19140704e-139,
       5.99548647e-138, 2.94432537e-136, 1.41105610e-134, 6.59933037e-133,
       3.01198334e-131, 1.34153612e-129, 5.83108375e-128, 2.47339397e-126,
       1.02384549e-124, 4.13592500e-123, 1.63045191e-121, 6.27249671e-120,
       2.35488659e-118, 8.62773226e-117, 3.08475324e-115, 1.07631974e-113,
       3.66487604e-112, 1.21779555e-110, 3.94899497e-109, 1.24967156e-107,
       3.85924456e-106, 1.16306991e-104, 3.42063276e-103, 9.81757406e-102,
       2.74978684e-100, 7.51607284e-099, 2.00484147e-097, 5.21874652e-096,
       1.32571288e-094, 3.28647139e-093, 7.95073620e-092, 1.87707635e-090,
       4.32467665e-089, 9.72349624e-088, 2.13347898e-086, 4.56826609e-085,
       9.54578226e-084, 1.94656434e-082, 3.87367436e-081, 7.52271421e-080,
       1.42568350e-078, 2.63674902e-077, 4.75895468e-076, 8.38207327e-075,
       1.44074939e-073, 2.41669888e-072, 3.95597690e-071, 6.31949042e-070,
       9.85161611e-069, 1.49875280e-067, 2.22510056e-066, 3.22378734e-065,
       4.55806207e-064, 6.28913898e-063, 8.46835892e-062, 1.11276749e-060,
       1.42694327e-059, 1.78569012e-058, 2.18073347e-057, 2.59893962e-056,
       3.02264330e-055, 3.43063659e-054, 3.79979038e-053, 4.10716045e-052,
       4.33232279e-051, 4.45961177e-050, 4.47992152e-049, 4.39178293e-048,
       4.20153934e-047, 3.92259174e-046, 3.57383825e-045, 3.17756016e-044,
       2.75708261e-043, 2.33454841e-042, 1.92909269e-041, 1.55560889e-040,
       1.22417877e-039, 9.40126677e-039, 7.04571412e-038, 5.15300741e-037,
       3.67784670e-036, 2.56167239e-035, 1.74120802e-034, 1.15498099e-033,
       7.47646185e-033, 4.72296269e-032, 2.91158864e-031, 1.75163100e-030,
       1.02837696e-029, 5.89195147e-029, 3.29429958e-028, 1.79748026e-027,
       9.57110925e-027, 4.97344736e-026, 2.52202789e-025, 1.24807117e-024,
       6.02734327e-024, 2.84059690e-023, 1.30644279e-022, 5.86365343e-022,
       2.56828549e-021, 1.09778024e-020, 4.57914744e-020, 1.86402165e-019,
       7.40481796e-019, 2.87061485e-018, 1.08600688e-017, 4.00947334e-017,
       1.44457200e-016, 5.07911637e-016, 1.74274659e-015, 5.83549130e-015,
       1.90685521e-014, 6.08072157e-014, 1.89229841e-013, 5.74673590e-013,
       1.70313848e-012, 4.92578884e-012, 1.39026864e-011, 3.82929452e-011,
       1.02928566e-010, 2.69991555e-010, 6.91132867e-010, 1.72651367e-009,
       4.20896778e-009, 1.00133245e-008, 2.32475979e-008, 5.26714152e-008,
       1.16457912e-007, 2.51281268e-007, 5.29112889e-007, 1.08726066e-006,
       2.18029929e-006, 4.26673490e-006, 8.14839952e-006, 1.51860937e-005,
       2.76195721e-005, 4.90213111e-005, 8.49082728e-005, 1.43519925e-004,
       2.36739920e-004, 3.81090307e-004, 5.98661620e-004, 9.17766154e-004,
       1.37302918e-003, 2.00458558e-003, 2.85605485e-003, 3.97105219e-003,
       5.38817575e-003, 7.13468852e-003, 9.21945801e-003, 1.16260683e-002,
       1.43072926e-002, 1.71822151e-002, 2.01371477e-002, 2.30310567e-002,
       2.57055488e-002, 2.79986444e-002, 2.97607750e-002, 3.08708491e-002]), 6: array([1.56250000e-02, 1.54354246e-02, 1.48803875e-02, 1.39993222e-02,
       1.28527744e-02, 1.15155283e-02, 1.00685738e-02, 8.59110757e-03,
       7.15364628e-03, 5.81303416e-03, 4.60972900e-03, 3.56734426e-03,
       2.69408787e-03, 1.98552610e-03, 1.42802743e-03, 1.00229279e-03,
       6.86514588e-04, 4.58883077e-04, 2.99330810e-04, 1.90545154e-04,
       1.18369960e-04, 7.17599624e-05, 4.24541364e-05, 2.45106555e-05,
       1.38097860e-05, 7.59304687e-06, 4.07419976e-06, 2.13336745e-06,
       1.09014964e-06, 5.43630328e-07, 2.64556444e-07, 1.25640634e-07,
       5.82289558e-08, 2.63357076e-08, 1.16237989e-08, 5.00666223e-09,
       2.10448389e-09, 8.63256833e-10, 3.45566434e-10, 1.34995778e-10,
       5.14642830e-11, 1.91464726e-11, 6.95134320e-12, 2.46289442e-12,
       8.51569240e-13, 2.87336795e-13, 9.46149206e-14, 3.04036078e-14,
       9.53427606e-15, 2.91774565e-15, 8.71373294e-16, 2.53955819e-16,
       7.22286001e-17, 2.00473667e-17, 5.43003438e-18, 1.43530742e-18,
       3.70240898e-19, 9.32010827e-20, 2.28957372e-20, 5.48890119e-21,
       1.28414274e-21, 2.93182672e-22, 6.53221396e-23, 1.42029845e-23,
       3.01367164e-24, 6.24035586e-25, 1.26101395e-25, 2.48672368e-26,
       4.78555463e-27, 8.98740128e-28, 1.64714979e-28, 2.94597574e-29,
       5.14188481e-30, 8.75815502e-31, 1.45579432e-31, 2.36148135e-32,
       3.73823093e-33, 5.77490495e-34, 8.70604008e-35, 1.28083619e-35,
       1.83892335e-36, 2.57650370e-37, 3.52285706e-38, 4.70063338e-39,
       6.12089383e-40, 7.77804443e-41, 9.64546346e-42, 1.16727420e-42,
       1.37854131e-43, 1.58878008e-44, 1.78691912e-45, 1.96129587e-46,
       2.10076967e-47, 2.19589147e-48, 2.23996076e-49, 2.22980589e-50,
       2.16616140e-51, 2.05358022e-52, 1.89989519e-53, 1.71531829e-54,
       1.51132165e-55, 1.29946981e-56, 1.09036673e-57, 8.92845058e-59,
       7.13471636e-60, 5.56383744e-61, 4.23417946e-62, 3.14456949e-63,
       2.27903104e-64, 1.61189367e-65, 1.11255028e-66, 7.49376401e-68,
       4.92580805e-69, 3.15974521e-70, 1.97798845e-71, 1.20834944e-72,
       7.20374693e-74, 4.19103663e-75, 2.37947734e-76, 1.31837451e-77,
       7.12841752e-79, 3.76135711e-80, 1.93683718e-81, 9.73282170e-83,
       4.77289113e-84, 2.28413306e-85, 1.06674346e-86, 4.87113350e-88,
       4.32467665e-89, 4.87113350e-88, 1.06674346e-86, 2.28413306e-85,
       4.77289113e-84, 9.73282170e-83, 1.93683718e-81, 3.76135711e-80,
       7.12841752e-79, 1.31837451e-77, 2.37947734e-76, 4.19103663e-75,
       7.20374693e-74, 1.20834944e-72, 1.97798845e-71, 3.15974521e-70,
       4.92580805e-69, 7.49376401e-68, 1.11255028e-66, 1.61189367e-65,
       2.27903104e-64, 3.14456949e-63, 4.23417946e-62, 5.56383744e-61,
       7.13471636e-60, 8.92845058e-59, 1.09036673e-57, 1.29946981e-56,
       1.51132165e-55, 1.71531829e-54, 1.89989519e-53, 2.05358022e-52,
       2.16616140e-51, 2.22980589e-50, 2.23996076e-49, 2.19589147e-48,
       2.10076967e-47, 1.96129587e-46, 1.78691912e-45, 1.58878008e-44,
       1.37854131e-43, 1.16727420e-42, 9.64546346e-42, 7.77804443e-41,
       6.12089383e-40, 4.70063338e-39, 3.52285706e-38, 2.57650370e-37,
       1.83892335e-36, 1.28083619e-35, 8.70604008e-35, 5.77490495e-34,
       3.73823093e-33, 2.36148135e-32, 1.45579432e-31, 8.75815502e-31,
       5.14188481e-30, 2.94597574e-29, 1.64714979e-28, 8.98740128e-28,
       4.78555463e-27, 2.48672368e-26, 1.26101395e-25, 6.24035586e-25,
       3.01367164e-24, 1.42029845e-23, 6.53221396e-23, 2.93182672e-22,
       1.28414274e-21, 5.48890119e-21, 2.28957372e-20, 9.32010827e-20,
       3.70240898e-19, 1.43530742e-18, 5.43003438e-18, 2.00473667e-17,
       7.22286001e-17, 2.53955819e-16, 8.71373294e-16, 2.91774565e-15,
       9.53427606e-15, 3.04036078e-14, 9.46149206e-14, 2.87336795e-13,
       8.51569240e-13, 2.46289442e-12, 6.95134320e-12, 1.91464726e-11,
       5.14642830e-11, 1.34995778e-10, 3.45566434e-10, 8.63256833e-10,
       2.10448389e-09, 5.00666223e-09, 1.16237989e-08, 2.63357076e-08,
       5.82289558e-08, 1.25640634e-07, 2.64556444e-07, 5.43630328e-07,
       1.09014964e-06, 2.13336745e-06, 4.07419976e-06, 7.59304687e-06,
       1.38097860e-05, 2.45106555e-05, 4.24541364e-05, 7.17599624e-05,
       1.18369960e-04, 1.90545154e-04, 2.99330810e-04, 4.58883077e-04,
       6.86514588e-04, 1.00229279e-03, 1.42802743e-03, 1.98552610e-03,
       2.69408787e-03, 3.56734426e-03, 4.60972900e-03, 5.81303416e-03,
       7.15364628e-03, 8.59110757e-03, 1.00685738e-02, 1.15155283e-02,
       1.28527744e-02, 1.39993222e-02, 1.48803875e-02, 1.54354246e-02]), 7: array([7.81250000e-03, 7.71771228e-03, 7.44019375e-03, 6.99966110e-03,
       6.42638721e-03, 5.75776417e-03, 5.03428691e-03, 4.29555378e-03,
       3.57682314e-03, 2.90651708e-03, 2.30486450e-03, 1.78367213e-03,
       1.34704394e-03, 9.92763048e-04, 7.14013714e-04, 5.01146394e-04,
       3.43257294e-04, 2.29441538e-04, 1.49665405e-04, 9.52725768e-05,
       5.91849800e-05, 3.58799812e-05, 2.12270682e-05, 1.22553278e-05,
       6.90489302e-06, 3.79652344e-06, 2.03709988e-06, 1.06668372e-06,
       5.45074822e-07, 2.71815164e-07, 1.32278222e-07, 6.28203171e-08,
       2.91144779e-08, 1.31678538e-08, 5.81189947e-09, 2.50333112e-09,
       1.05224194e-09, 4.31628416e-10, 1.72783217e-10, 6.74978888e-11,
       2.57321415e-11, 9.57323631e-12, 3.47567160e-12, 1.23144721e-12,
       4.25784620e-13, 1.43668398e-13, 4.73074603e-14, 1.52018039e-14,
       4.76713803e-15, 1.45887282e-15, 4.35686647e-16, 1.26977909e-16,
       3.61143000e-17, 1.00236834e-17, 2.71501719e-18, 7.17653712e-19,
       1.85120449e-19, 4.66005414e-20, 1.14478687e-20, 2.74445105e-21,
       6.42073764e-22, 1.46603769e-22, 3.27241205e-23, 7.41351003e-24,
       3.01367164e-24, 7.41351003e-24, 3.27241205e-23, 1.46603769e-22,
       6.42073764e-22, 2.74445105e-21, 1.14478687e-20, 4.66005414e-20,
       1.85120449e-19, 7.17653712e-19, 2.71501719e-18, 1.00236834e-17,
       3.61143000e-17, 1.26977909e-16, 4.35686647e-16, 1.45887282e-15,
       4.76713803e-15, 1.52018039e-14, 4.73074603e-14, 1.43668398e-13,
       4.25784620e-13, 1.23144721e-12, 3.47567160e-12, 9.57323631e-12,
       2.57321415e-11, 6.74978888e-11, 1.72783217e-10, 4.31628416e-10,
       1.05224194e-09, 2.50333112e-09, 5.81189947e-09, 1.31678538e-08,
       2.91144779e-08, 6.28203171e-08, 1.32278222e-07, 2.71815164e-07,
       5.45074822e-07, 1.06668372e-06, 2.03709988e-06, 3.79652344e-06,
       6.90489302e-06, 1.22553278e-05, 2.12270682e-05, 3.58799812e-05,
       5.91849800e-05, 9.52725768e-05, 1.49665405e-04, 2.29441538e-04,
       3.43257294e-04, 5.01146394e-04, 7.14013714e-04, 9.92763048e-04,
       1.34704394e-03, 1.78367213e-03, 2.30486450e-03, 2.90651708e-03,
       3.57682314e-03, 4.29555378e-03, 5.03428691e-03, 5.75776417e-03,
       6.42638721e-03, 6.99966110e-03, 7.44019375e-03, 7.71771228e-03]), 'xi': 0.0, 'sigma': 0.000390625, 'j': 0})
```

**Comment by entn-at** [comment]
For anyone running into this issue with TF/Keras and wants to use TF>=2.5, I made some fixes in a local branch: https://github.com/entn-at/kymatio/tree/fix-dict-wrapper

The changes are essentially to use `str` as keys for `phi_f` and `psi` dicts instead of mixing `int`s and `str`s. Unfortunately, this then breaks torch backend tests (numpy etc. still work). As I currently don't have time to look into this further (and I'm not sure if this will be relevant for future/dev versions), I'll leave this be for now.

**Comment by MuawizChaudhary** [comment]
This is a pretty serious issue and somewhat pops up in #801. We should triage this as high priority. 

To quote my response there; 

"[On my end](https://colab.research.google.com/drive/16qJ0f5-G6_FPKgo99Hf39GrH0QznC3pn?usp=sharing), it seems to pass line 145 (I tested with your J, T, Q, and also with smaller.). Furthermore, like you, I see nothing about an agnostic backend. Perhaps this was added and then removed. Perhaps you were using https://github.com/kymatio/kymatio/pull/738 ?

It does error out with the following when you try summarizing our scattering object...

ValueError: Error processing property 'phi_f' of DictWrapper({0: array(....
And it does appear to make alot of function calls when its running...
[image](https://user-images.githubusercontent.com/39755015/171431232-93d7121d-225c-4ddd-8178-bda940d642e8.png)

Whose our local Keras expert here?:

**Comment by edouardoyallon** [comment]
I did the keras backend but I don't know anything about this software..

**Comment by MuawizChaudhary** [comment]
@edouardoyallon I'm pro putting up a issue on the tensorflow/keras github, I think this might be a little out of scope for the usual keras users even...

If no one objects, I'll make an issue later tonight or tomorrow morning.

**Comment by janden** [comment]
Agree that this is high priority.

> For anyone running into this issue with TF/Keras >=2.5, I made some fixes in a local branch: https://github.com/entn-at/kymatio/tree/fix-dict-wrapper
> 
> The changes are essentially to use `str` as keys for `phi_f` and `psi` dicts instead of mixing `int`s and `str`s. Unfortunately, this then breaks torch backend tests (numpy etc. still work). As I currently don't have time to look into this further (and I'm not sure if this will be relevant for future/dev versions), I'll leave this be for now.

This makes sense. It's not clear why we should be mixing ints and strings as keys here in the first place. Will see if there's a quick fix we can make.

**Comment by lostanlen** [comment]
> . It's not clear why we should be mixing ints and strings as keys here in the first place

Mixing ints and strings is incredibly useful here.
The ints correspond to the "resolutions" at which filters are discretized (i.e., periodized in Fourier). This is not needed at the first layer but speeds up computation a lot at the second layer, because the psi2's operate over a subsampled input.
Same when applying phi's onto a multiresolution U layer to get a single-resolution S layer. 

Meanwhile, the strings are metadata tokens: `xi` (center frequency), `sigma` (bandwidth), and `j` (output resolutions). These tokens are important for padding, sampling, depth first search, precomputing the size, and returning metadata for each scattering path.

**Comment by janden** [comment]
> Mixing ints and strings is incredibly useful here.

I agree that both are needed, but there's probably a more logical structure. A first step would be to put the actual filters in a sub-list (say `psi1_f["filters"][j]` instead of `psi1_f[j]`). This should fix the TF issue without the slowdowns that @entn-at is describing. Really, though, these metadata should probably be stored in a separate dict from the filters.

**Comment by lostanlen** [comment]
`"filters"` is not great as a key name. It's really the same filter, just expressed at a different resolution.
Other name ideas:
```python
data
levels
scales
values
```

i like `levels`

**Comment by MuawizChaudhary** [comment]
This should be fixed with the mergal of #882.
